tck/src/main/java/org/eclipse/microprofile/lra/tck/TckParticipantTests.java

```diff
@@ -169,7 +180,42 @@ public void testNonJaxRsCompletionStageResponseAndParticipantStatus() throws Int
 
         assertTrue("Non JAX-RS @Status method with CompletionStage<ParticipantStatus> should have been called",
             lraMetricService.getMetric(LRAMetricType.Status, lraId) >= 1);
-        
+
         lraTestService.waitForRecovery(lraId);
     }
+
+    @Test
+    public void cancelLraDuringBusinessMethod() throws InterruptedException, ExecutionException {
+        ExecutorService es = Executors.newSingleThreadExecutor();
+        LRAClientOps lraOps = lraTestService.getLRAClient();
+        URI lraId = lraOps.startLRA(null, lraClientId(), 0L, ChronoUnit.MILLIS);
```

Comment:
Would it be possible to also have a test where the spec implementation starts the LRA:
``` java
@LRA(value = LRA.Type.REQUIRES_NEW)
```

So it would be a similar test but without the explicit cancel: `lraOps.cancelLRA(lraId);`.

Reply:
I'm not quite sure whether this test will be still similar to `cancelLraDuringBusinessMethod`. Since we don'ŧ externally cancel the LRA, do we still expect `@Compensate` to be executed? Were you thinking about incorporating timeout in some way to the additional test?

Reply:
I believe that is exactly what @mmusgrov is talking about. However, I don't think an additional test is necessary as it would be just additionally checking that timelimit is fired and we have a test for that elsewhere.